### PR TITLE
OKTA-295706 add usertypes

### PR DESCRIPTION
**Pull request by bryanapellanes-okta**
## Issue \#
OKTA-295706

## Code

- [x] Integration test(s)
- [x] Implementation

### CODE REVIEW

File: src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs [hunk 1]
```diff
@@ -372,11 +372,12 @@ public async Task AddOpenIdConnectApp()
         {
             var client = TestClient.Create();
             var guid = Guid.NewGuid();
+            var testClientId = $"{nameof(AddOpenIdConnectApp)}_TestClientId";
 
             var createdApp = await client.Applications.CreateApplicationAsync(new CreateOpenIdConnectApplication
             {
                 Label = $"dotnet-sdk: AddOpenIdConnectApp {guid}",
-                ClientId = "0oae8mnt9tZcGcMXG0h3",
+                ClientId = testClientId,
```
Comment (laura-rodriguez) [inline src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs:380]: Why this change is needed?

Comment (bryanapellanes-okta) [reply src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs:380]: This was causing integration tests to fail because the same value was being used by multiple tests.  There was a 400 returned by the server because an object with the same clientId already exist.

Comment (bryanapellanes-okta) [reply src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs:380]: I could move this change into a separate branch/set of changes so usertypes is isolated if you prefer.

Comment (laura-rodriguez) [reply src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs:380]: oh right! this is the one you mentioned today. It's Ok, we can keep it here.

File: src/Okta.Sdk.IntegrationTests/UserTypesClientShould.cs [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+                    Description = updatedDescription,
+                    DisplayName = updatedDisplayName,
+                };
+
+                var updatedUserType = await testClient.UserTypes.UpdateUserTypeAsync(updatesToUserType, createdUserType.Id);
+                updatedUserType.Id.Should().Be(createdUserType.Id);
+                updatedUserType.Description.Should().Be(updatedDescription);
+                updatedUserType.DisplayName.Should().Be(updatedDisplayName);
+
+                var retrievedForValidation = await testClient.UserTypes.GetUserTypeAsync(createdUserType.Id);
```
Comment (laura-rodriguez) [inline src/Okta.Sdk.IntegrationTests/UserTypesClientShould.cs:119]: nit: I suggest using more descriptive names. I would rename this to `retrievedUserTypeForValidation`, so we know it is a user type.

File: src/Okta.Sdk.IntegrationTests/UserTypesClientShould.cs [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+                        DisplayName = replacedDisplayName,
+                        Name = testName,
+                    }, createdUserType.Id);
+
+                replacedUserType.Id.Should().Be(createdUserType.Id);
+                replacedUserType.Description.Should().Be(replacedDescription);
+                replacedUserType.DisplayName.Should().Be(replacedDisplayName);
+                replacedUserType.Name.Should().Be(testName);
+
+                var retrievedForValidation = await testClient.UserTypes.GetUserTypeAsync(createdUserType.Id);
```
Comment (laura-rodriguez) [inline src/Okta.Sdk.IntegrationTests/UserTypesClientShould.cs:165]: nit: descriptive names. See above.

File: src/Okta.Sdk.IntegrationTests/UserTypesClientShould.cs [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+                    Description = $"{nameof(ListAllUserTypes)} Test Description ({i})",
+                    DisplayName = $"{nameof(ListAllUserTypes)} Test DisplayName ({i})",
+                    Name = $"{nameof(ListAllUserTypes)}_TestUserType_{i}",
+                });
+                testUserTypeIds.Add(createdUserType.Id);
+            }
+
+            try
+            {
+                HashSet<string> allUserTypeIds = new HashSet<string>();
```
Comment (laura-rodriguez) [inline src/Okta.Sdk.IntegrationTests/UserTypesClientShould.cs:235]: nit: In this method we have vars and explicitly typed vars, let's try to use the same coding rules when it's possible for consistency. In general, this SDK is using vars, so I would stick to it.
